### DISCUSSION TITLE
is a fenced code block meant to become a <pre><code> block?

### DISCUSSION THREAD

**Discussion by Southclaws** [mdx-js/mdx#2415] [Q&A]
surely it should just be \<pre\> I've never seen \<code\> nested within, now I've gotta mess around with styles that only affect a lone \<code\> separate from one inside a \<pre\>

**Comment by wooorm** [comment]
Your wrong. Every tool does this. And its in the markdown spec. The code element is for code.

**Comment by ChristianMurphy** [comment]
Welcome @Southclaws! 👋 
Sorry you ran into a spot of confusion.

mdx, remark, and most markdown parsers follow CommonMark (https://commonmark.org/).
In CommonMark, fenced codeblocks include both `<pre>` and `<code>`, that is standard and there is no intent to change that as a default. https://spec.commonmark.org/0.30/#fenced-code-blocks

Styling is the way to go.

If you want/need to change the output structure rather than styling.
It is possible though plugins https://unifiedjs.com/learn/
Though is significantly more involved than adding a couple styles.
